lib/covenants/namestate.js

```diff
@@ -740,6 +741,14 @@ class NameState extends bio.Struct {
 
     const stats = {};
 
+    if (this.isExpired(height, network)) {
```

Comment:
I don't FULLY understand all the state transitions but because this `isExpired` is preceding everything I was wondering if all checks do make sense from isExpired.

isExpired has three `true` returns, one of them is:
   1. When name is revoked and then expires (revoked + auction maturity) - Don't we want to include isRevoked information here as well ? (to provide stats whether it was revoked -> expired vs someone forgot to renew?)
   2. If we have not renewed in two years - I guess this PR is concentrated on that case.
   3. Auction failed - no one revealed their bids.
-- Results?
  1. I guess `expired` would be `this.renewal + network.names.auctionMaturity` or `this.revoked + network.names.auctionMaturity` (My understanding, after looking at chain code, is `this.renewal` is always updated when something changes about a name)
  2. I believe this is the case, when this `expire` value makes sense, it just expired w/o renewal.
  3. In this case, I am not sure if we even call this expire. `expired` would probably be the `this.renewal` itself? (last name update when reveal failed?)

Reply:
Ok I think you're right I need to add a test for `revoked` and make sure this still makes sense. The revoked period is `auctionMaturity` after which the name is just `CLOSED` and can be reopen for a brand new auction. The point of this PR is that after a name is expired (which is `this.renewal + network.names.renewalWindow`) the JSON stats will indicate a `CLOSED` state but with negative numbers for the renewal, which is invalid after expiration:

```
  stats: {
    renewalPeriodStart: 72,
    renewalPeriodEnd: 5072,
    blocksUntilExpire: -4928,
    daysUntilExpire: -34.22
  }
```

...and makes even less sense if the name was in mid-transfer (this was broken recently in https://github.com/handshake-org/hsd/pull/548)

```
  stats: {
    renewalPeriodStart: 72,
    renewalPeriodEnd: 5072,
    blocksUntilExpire: 1,
    daysUntilExpire: 0.01,
    transferLockupStart: 98,
    transferLockupEnd: 108,
    blocksUntilValidFinalize: -4963,
    hoursUntilValidFinalize: -827.17
  }
```

Note: `this.renewal` is not always updated when something changes about a name. You can see in `chain.verifyCovenants()` that `ns.setRenewal()` is only called on certain covenants: `CLAIM`, `REGISTER`, `RENEW` and `FINALIZE`

Reply:
What about failed auction (Even though it should not happen that often) ? - Case when no one revealed.
OPEN -> BID(s) and no one reveals after that. (I believe that's the last case I mentioned)

Reply:
You could just use custom version of isExpired - `if (isExpired(..) && !isRevoked(..) && !this.owner.isNull())` to target this specific case. (Or create another helper method)

Reply:
An auction that ends with no reveals goes back to null state:

```
{
  "start": {
    "reserved": false,
    "week": 0,
    "start": 0
  },
  "info": null
}
```